src/tests/ftest/osa/osa_offline_extend.py

```diff
@@ -22,83 +24,190 @@ def setUp(self):
+        self.test_oclass = None
+        self.dmg_command.exit_status_exception = True
 
-    def run_offline_extend_test(self, num_pool, data=False):
+    def run_offline_extend_test(self, num_pool, data=False,
+                                server_boot=False, oclass=None):
         """Run the offline extend without data.
 
         Args:
             num_pool (int) : total pools to create for testing purposes.
             data (bool) : whether pool has no data or to create
                           some data in pool. Defaults to False.
+            server_boot (bool) : Perform system stop/start on a rank.
+                                 Defaults to False.
+            oclass (list) : list of daos object class (eg: "RP_2G8")
         """
         # Create a pool
         pool = {}
-        pool_uuid = []
+        tmp = server_boot
+        self.log.info(tmp)
+        if oclass is None:
+            oclass = []
+            oclass.append(self.ior_cmd.dfs_oclass.value)
 
-        # Extend a ranks 4 and 5
-        rank = [4, 5]
+        self.log.info(oclass[0])
+        # tmpfile = self.ior_cmd.test_file.value
 
         for val in range(0, num_pool):
+            # Perform IOR write using the oclass list
+            if val < len(oclass):
+                index = val
+            else:
+                index = 0
+            # tstring = "{}".format(val)
+            # self.ior_cmd.test_file.value = tmpfile.join([tstring])
             pool[val] = TestPool(self.context, dmg_command=self.dmg_command)
             pool[val].get_params(self)
-            # Split total SCM and NVME size for creating multiple pools.
-            pool[val].scm_size.value = int(pool[val].scm_size.value /
-                                           num_pool)
-            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
-                                            num_pool)
             pool[val].create()
-            pool_uuid.append(pool[val].uuid)
             self.pool = pool[val]
+            test_seq = self.ior_test_sequence[0]
+            self.pool.set_property("reclaim", "disabled")
             if data:
-                self.write_single_object()
+                self.run_ior_thread("Write", oclass[index], test_seq)
+                self.run_mdtest_thread()
+                if self.test_during_aggregation is True:
+                    self.run_ior_thread("Write", oclass[index], test_seq)
         # Start the additional servers and extend the pool
         self.log.info("Extra Servers = %s", self.extra_servers)
         self.start_additional_servers(self.extra_servers)
         # Give sometime for the additional server to come up.
-        time.sleep(5)
+        for retry in range(0, 10):
+            scan_info = self.get_dmg_command().system_query()
+            if not check_system_query_status(scan_info):
+                if retry == 9:
+                    self.fail("One or more servers not in expected status")
```

Comment:
I thought you can break the for loop if check_system_query_status() return true?

Reply:
Yes, you are right... Will fix it...Thanks.

Reply:
Done.